src/child/components/Favourite.js

```diff
@@ -76,11 +91,33 @@ class Favourite extends Component {
         }
     }
 
+    onModalClick(e) {
```

Comment:
I'm not sure I'd put this in the favourite component and make it a prop for the confirmation component, if all it's doing is stopping the event propagating.